packages/lib/src/textarea/Textarea.tsx

```diff
@@ -1,66 +1,178 @@
+import { HalstackLanguageContext } from "../HalstackContext";
 import TextareaPropsType, { RefType } from "./types";
+import { scrollbarStyles } from "../styles/scroll";
+import DxcIcon from "../icon/Icon";
+
+const sizes = {
+  small: "240px",
+  medium: "360px",
+  large: "480px",
+  fillParent: "100%",
+};
+
+const calculateWidth = (margin: TextareaPropsType["margin"], size: TextareaPropsType["size"]) =>
+  size === "fillParent"
+    ? `calc(${sizes[size]} - ${getMargin(margin, "left")} - ${getMargin(margin, "right")})`
+    : size && sizes[size];
+
+const TextareaContainer = styled.div<{
+  margin: TextareaPropsType["margin"];
+  size: TextareaPropsType["size"];
+}>`
+  display: flex;
+  flex-direction: column;
+  width: ${({ margin, size }) => calculateWidth(margin, size)};
+  margin: ${({ margin }) => (margin && typeof margin !== "object" ? spaces[margin] : "0px")};
+  margin-top: ${({ margin }) => (margin && typeof margin === "object" && margin.top ? spaces[margin.top] : "")};
+  margin-right: ${({ margin }) => (margin && typeof margin === "object" && margin.right ? spaces[margin.right] : "")};
+  margin-bottom: ${({ margin }) =>
+    margin && typeof margin === "object" && margin.bottom ? spaces[margin.bottom] : ""};
+  margin-left: ${({ margin }) => (margin && typeof margin === "object" && margin.left ? spaces[margin.left] : "")};
+`;
+
+const Label = styled.label<{
+  disabled: TextareaPropsType["disabled"];
+  helperText: TextareaPropsType["helperText"];
+}>`
+  color: ${({ disabled }) => (disabled ? "var(--color-fg-neutral-medium)" : "var(--color-fg-neutral-dark)")};
+  font-family: var(--typography-font-family);
+  font-size: var(--typography-label-m);
+  font-weight: var(--typography-label-semibold);
+  ${({ helperText }) => !helperText && "margin-bottom: var(--spacing-gap-xs);"}
+
+  /* Optional text */
+  > span {
+    color: ${({ disabled }) => (disabled ? "var(--color-fg-neutral-medium)" : "var(--color-fg-neutral-stronger)")};
+    font-weight: var(--typography-label-regular);
+  }
+`;
+
+const HelperText = styled.span<{ disabled: TextareaPropsType["disabled"] }>`
+  color: ${({ disabled }) => (disabled ? "var(--color-fg-neutral-medium)" : "var(--color-fg-neutral-stronger)")};
+  font-family: var(--typography-font-family);
+  font-size: var(--typography-helper-text-s);
+  font-weight: var(--typography-helper-text-regular);
+  margin-bottom: var(--spacing-gap-xs);
+`;
+
+const Textarea = styled.textarea<{
+  error: TextareaPropsType["error"];
+  verticalGrow: TextareaPropsType["verticalGrow"];
+}>`
+  ${({ verticalGrow }) => {
+    if (verticalGrow === "none") return "resize: none;";
+    else if (verticalGrow === "auto") return `resize: none; overflow: hidden;`;
+    else if (verticalGrow === "manual") return "resize: vertical;";
+    else return `resize: none;`;
+  }};
+  ${scrollbarStyles}
+  padding: var(--spacing-padding-xs) var(--spacing-padding-xs) var(--spacing-padding-xxxs) var(--spacing-padding-xs);
```

Comment:
Padding-right should also be var(--spacing-padding-xxxs).

Reply:
Discussed with the designers, we keep it like that.